web/themes/new_weather_theme/assets/js/components/TabbedNavigator.js

```diff
@@ -0,0 +1,146 @@
+class TabbedNavigator extends HTMLElement {
+  constructor() {
+    super();
+
+    // Bind this context to methods that need it
+    this.handleAlertAnchorClick = this.handleAlertAnchorClick.bind(this);
+    this.handleTabButtonClick = this.handleTabButtonClick.bind(this);
+    this.switchToTab = this.switchToTab.bind(this);
+    this.scrollToAccordion = this.scrollToAccordion.bind(this);
+  }
+
+  connectedCallback() {
+    // If no tabs are selected by default, then select the first one
+    const selected = Array.from(
+      this.querySelectorAll(".tab-button[data-selected]"),
+    );
+    if (!selected.length) {
+      this.switchToTab(this.querySelector("button").dataset.tabName);
+    }
+
+    // The initial page load might contain a hash fragment
+    // referring either to content within a given tab
+    // (such as an alert) or a tab itself. We should handle
+    // these two cases.
+    this.navigateWithInitialHash();
+
+    // Intercept click events on Alert links at the
+    // top of the page and handle them in this component
+    Array.from(document.querySelectorAll("weathergov-alert-list a")).forEach(
+      (alertAnchor) => {
+        alertAnchor.addEventListener("click", this.handleAlertAnchorClick);
+      },
+    );
+
+    // Add needed event listeners
+    Array.from(this.querySelectorAll("button.tab-button")).forEach((button) => {
+      button.addEventListener("click", this.handleTabButtonClick);
+    });
+  }
+
+  disconnectedCallback() {
+    // Remove any event listeners
+    Array.from(this.querySelectorAll("button.tab-button")).forEach((button) => {
+      button.removeEventListener("click", this.handleTabButtonClick);
+    });
+  }
+
+  navigateWithInitialHash() {
+    const hash = new URL(window.location).hash;
+    if (!hash || hash === "") {
+      return;
+    }
+
+    const matchedTabButton = this.querySelector(
+      `[data-tab-name="${hash.replace("#", "")}"]`,
+    );
+    if (matchedTabButton) {
+      this.switchToTab(matchedTabButton.dataset.tabName);
+      matchedTabButton.parentElement.scrollIntoView();
+      return;
+    }
+
+    const childElement = this.querySelector(`${hash}`);
+    if (childElement) {
+      const tabContainer = childElement.closest(".tab-container");
+      this.switchToTab(tabContainer.id);
```

Comment:
What happens if the hash refers to an element that isn't side our tab containers? Would it be worth adding an `if(tabContainer)` ahead of switching tabs, just to be safe?

Reply:
The only other option is for it to be a tab button, which we already deal with in the case(s) above. But -- if we change the structure of the component children then you are right, it could present a problem. We might also want to actually enforce a shadow dom and slots, ensuring that we _don't_ display anything that isn't a button area or a tab container...

Reply:
Yeah, fair enough. Could it pop up if again if we add hash navigation elsewhere on the page? In any case, no need to deal with it right now.